pandas/core/frame.py

```diff
@@ -3924,16 +3924,18 @@ def _set_value(
             Sets whether or not index/col interpreted as indexers
         """
         try:
+            # setitem will do validation that may raise TypeError,
```

Comment:
should this comment refer to column_setitem instead of just setitem?

Reply:
it also looks like the new method doesn't actually do the validation this comment refers to?

Reply:
This is an existing comment (originally a few lines below), but so I _suppose_ this comment was actually already not up to date anymore. 
So before this PR the comment was about `setitem_inplace`, and that also doesn't do any validation.

Reply:
setitem_inplace calls np_can_hold_element, which raises on failure

Reply:
Ah, yes, I see. That is doing validation that can raise the LossySetitemError. So the new `column_setitem` calls `setitem`, which will also call `np_can_hold_element`, but there catching the LossySetitemError and coercing to the target dtype if needed. 
That is something that the loc/iloc fallback below otherwise will also do, so I _suppose_ this change is OK (but the comment is then indeed no longer correct, and we also don't need to catch LossySetitemError here)

Reply:
> and we also don't need to catch LossySetitemError here

yah, possibly also TypeError and ValueError